core/include/scipp/core/parallel-tbb.h

```diff
@@ -27,7 +27,12 @@ inline auto blocked_range(const scipp::index begin, const scipp::index end,
 }
 
 template <class... Args> void parallel_for(Args &&... args) {
-  tbb::parallel_for(std::forward<Args>(args)...);
+  if (auto *thread_limit = std::getenv("SCIPP_NUM_THREADS")) {
```

Comment:
`parallel_for` is in some cases used in nested contexts. I would fear that having this here will severely limit performance.

Have you tried? Enable this check only with an optional compile-time flag?

Reply:
I didn't note any loss of performance, but I didn't try in a nested context. Applying via a CMake flag would probably be better anyway.